### ISSUE TITLE
Error upon submitting new package

### ISSUE THREAD

**Issue by duarteroso** [vlang/vpm#126]
### Describe the bug

Whenever I attempt to sbmit a new package, the following error appears at the top:

```pg orm_stmt_worker error: ERROR: duplicate key value violates unique constraint "User_username_key" DETAIL: Key (username)=() already exists.```

![image](https://github.com/user-attachments/assets/21315ae7-0356-40c4-a039-837bb3b5c130)

### Expected Behavior

The packaged should be added to the list

### Current Behavior

An error prevents the package from being added

### Reproduction Steps

Not sure if any package or just mine.
Attempt to add a new package and hopefully this error will appear for eveybody

### V version

V 0.4.8

### Environment details (OS name and version, etc.)

Irrelevant

**Comment by KeyWeeUsr** [comment]
@medvednikov ping

**Comment by medvednikov** [comment]
Having a look

**Comment by koki-develop** [comment]
I encountered the same issue and after investigating, I was able to reproduce it locally.
Here's my report:

---

First, when logging in for the first time, a record is created in the `User` table:

```
vpm=# select * from "User";
 id | github_id |   username   | avatar_url | is_blocked | block_reason | is_admin |                                              random_id                                               |     created_at      |     updated_at
----+-----------+--------------+------------+------------+--------------+----------+------------------------------------------------------------------------------------------------------+---------------------+---------------------
  1 |         0 | koki-develop |            | f          |              | f        | [`x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`] | 2025-01-05 20:31:11 | 2025-01-05 20:31:11
(1 row)
```

Then, when submitting a package, a record is created in the `Package` table as expected:

```
vpm=# select * from "Package";
 id |         name         | description | documentation |                    url                    | nr_downloads | vcs | user_id | author_id | stars | is_flatten |     updated_at      |     created_at
----+----------------------+-------------+---------------+-------------------------------------------+--------------+-----+---------+-----------+-------+------------+---------------------+---------------------
  1 | koki-develop.spinner | test        |               | https://github.com/koki-develop/v-spinner |            0 | git |       1 |         2 |     0 | f          | 2025-01-05 20:33:36 | 2025-01-05 20:33:36
(1 row)
```

Interestingly, when checking the `User` table again, **a second record has been created**. What's notable is that this record has **an empty string (`''`) as its `username`**:

```
vpm=# select * from "User";
 id | github_id |   username   | avatar_url | is_blocked | block_reason | is_admin |                                              random_id                                               |     created_at      |     updated_at
----+-----------+--------------+------------+------------+--------------+----------+------------------------------------------------------------------------------------------------------+---------------------+---------------------
  1 |         0 | koki-develop |            | f          |              | f        | [`x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`, `x`] | 2025-01-05 20:31:11 | 2025-01-05 20:31:11
  2 |         0 |              |            | f          |              | f        |                                                                                                      | 2025-01-05 20:33:36 | 2025-01-05 20:33:36
(2 rows)
```

And when trying to submit a second package, **an error occurs**:

```
pg orm_stmt_worker error: ERROR: duplicate key value violates unique constraint "User_username_key" DETAIL: Key (username)=() already exists.
```

This error occurs in the package insertion code:
https://github.com/vlang/vpm/blob/d81991115af5367cdd8443e09c679efdff04f654/src/repo/package.v#L116-L118

It appears that when inserting a `Package`, a `User` record is also being inserted for some reason, with an empty username.

The `User` table has a unique constraint on the `username` field:
https://github.com/vlang/vpm/blob/d81991115af5367cdd8443e09c679efdff04f654/src/entity/user.v#L10

---

To summarize the sequence of events:

1. Submit a package
   - This creates a `User` record with an empty username
2. Try to submit another package
   - Try again to create a `User` record with an empty username
   - Due to the unique constraint on `username`, and because a record with an empty username already exists, an error occurs

---

For reference, here's the `Package` struct definition:
https://github.com/vlang/vpm/blob/d81991115af5367cdd8443e09c679efdff04f654/src/entity/package.v#L6-L21

Could it be that when inserting a `Package`, the `author` field (of type `User`) is also being inserted? 🤔
(I'm not familiar with vlang's ORM behavior, so I'm not sure about the exact mechanics here.)

I'll look into it a bit more when I have time!
